lib/hydra/pcdm/models/concerns/collection_behavior.rb

```diff
@@ -50,50 +64,24 @@ def parent_collections
       aggregated_by.select { |parent| parent.class.included_modules.include?(Hydra::PCDM::CollectionBehavior) }
     end
 
-    def child_collections= collections
-      raise ArgumentError, "each collection must be a pcdm collection" unless collections.all? { |c| Hydra::PCDM.collection? c }
-      raise ArgumentError, "a collection can't be an ancestor of itself" if collection_ancestor?(collections)
-      self.members = child_objects + collections
-    end
-
-    def child_collections
-      members.to_a.select { |m| Hydra::PCDM.collection? m }
-    end
-
     def child_collection_ids
       child_collections.map(&:id)
     end
 
     include ChildObjects
 
-    def child_objects= objects
-      raise ArgumentError, "each object must be a pcdm object" unless objects.all? { |o| Hydra::PCDM.object? o }
-      self.members = child_collections + objects
-    end
-
-
-    def collection_ancestor? collections
-      collections.each do |check|
-        return true if ancestor?(check)
+     def collection_ancestor? collections
+      warn "[DEPRECATION] `collection_ancestor?` is deprecated.  Please use `AncestorChecker.new(parent_collection).ancestor?(child_collection)` for each collection instead.  This has a target date for removal of 07-31-2015"
+      collections.each do |col|
+        return true if AncestorChecker.new(self).ancestor?(col)
       end
       false
     end
 
     def ancestor? collection
-      return true if collection == self
-      return false if collection.child_collections.empty?
-      current_collections = collection.child_collections
-      next_batch = []
-      while !current_collections.empty? do
-        current_collections.each do |c|
-          return true if c == self
-          next_batch += c.child_collections
-        end
-        current_collections = next_batch
-      end
-      false
+      warn "[DEPRECATION] `collection_ancestor?` is deprecated.  Please use `AncestorChecker.new(parent_collection).ancestor?(child_collection)` instead.  This has a target date for removal of 07-31-2015"
```

Comment:
@terrellt, I also thought about keeping ancestor? as part of the API with it calling the AncestorChecker.  It could be updated to allow for one or an array of children, which would collapse the existing two methods into one.  This is consistent with the way methods like delete work.  If you want that, can you open a separate issue?